replication project.Rmd

```diff
@@ -0,0 +1,259 @@
+variable <- salience
+slopes <- model$coef$fixed["var_c"] + model$coef$fixed["salience_mean_c:var_c"] * variable # Estimated slopes based on estimated coefficients of salience (21) and salience x variance (39)
+estmean <- model$coef$fixed # Estimated means for variables 
+estvar <- vcov(model) # Estimated covariances
+se <- rep(NA, length(variable)) # Creating empty vector for predicted s.e. values
+# Creating a loop with deltamethod to calculate estimated standard errors
+for (i in 1:length(variable)) {
+  j <- variable[i]
+  se[i] <- deltamethod (~ (x21) + (x39)*j, estmean, estvar)
+}
+# Create confidence intervals and combine all values in matrix.
+upper <- slopes + 1.96*se
+lower <- slopes - 1.96*se
+variable.data <- cbind(variable, slopes, upper, lower)
+
+# Creating the first graph with ggplot:
+
+variable.data <- data.frame(variable.data) # Turn matrix into data frame so that ggplot can read it.
+xtitle <- "Salience"
+ytitle <- "Effect: Variance"
+main <- "The Conditional Effects of Salience and Variance"
+graph_3 <- ggplot(variable.data, aes(x = variable, y = slopes)) + 
+  geom_line(aes(y = slopes), color = "blue") + 
+  geom_ribbon(aes(ymin = lower, ymax = upper), alpha = 0.2) + 
+  geom_hline(yintercept = 0, size = 0.4, color = "black") +
+  labs(x = xtitle, y = ytitle, title = main) +
+  coord_fixed(ratio = 400)
+
+# Second part of Figure 2:
+
+variance <- seq(from = min(data$var_c), to = max(data$var_c), by=10) # vector with possible values for variance
+variable <- variance
+slopes <- model$coef$fixed["salience_mean_c"] + model$coef$fixed["salience_mean_c:var_c"] * variable # Estimated slopes based on estimated coefficients of salience (20) and salience x variance (39)
+estmean <- model$coef$fixed # Estimated means for variables 
+estvar <- vcov(model) # Estimated covariances
+se <- rep(NA, length(variable)) # Creating empty vector for predicted s.e. values
+# Creating a loop with deltamethod to calculate estimated standard errors
+
+for (i in 1:length(variable)) {
+  j <- variable[i]
+  se[i] <- deltamethod (~ (x20) + (x39) * j, estmean, estvar)
+}
+# Create confidence intervals and combine all values in matrix.
+upper <- slopes + 1.96 * se
+lower <- slopes - 1.96 * se
+variable.data <- cbind(variable, slopes, upper, lower)
+
+# Creating the second graph with ggplot:
+
+variable.data <- data.frame(variable.data) # Turn matrix into data frame so that ggplot can read it.
+xtitle <- "Variance"
+ytitle <- "Effect: Salience"
+main <- "The Conditional Effects of Salience and Variance"
+graph_4 <- ggplot(variable.data, aes(x = variable, y = slopes)) + 
+  geom_line(aes(y = slopes), color = "blue") + 
+  geom_ribbon(aes(ymin = lower, ymax = upper), alpha = 0.2) + 
+  geom_hline(yintercept = 0, size = 0.4, color = "black") +
+  labs(x = xtitle, y = ytitle, title = main) +
+  coord_fixed(ratio = 500)
+
+grid.arrange(graph_3, graph_4, ncol = 1)
+# Better scaled versions - not combined
+graph_3
+graph_4
+```
+
```

Comment:
Overall, the code is clear and well commented.  The next big step to improve the code, should you choose to do it, would be turn this into one function that is called several times.